File: Olympiad/COCI/official/2007/contest1_solutions/okviri.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 1 - Task OKVIRI
  Programming language Pascal
*)

program okviri;

var
   ulaz       : string;
   i, n, r, s : integer;
   izlaz      : array[1..5] of string;

procedure crtaj(x: char; c: integer; y: char);
var dr, ds : integer;
begin
   izlaz[3][c] := x;
   for dr:=-2 to 2 do
      for ds:=-2 to 2 do
         if abs(dr) + abs(ds) = 2 then
            izlaz[3+dr][c+ds] := y;
end;

begin
   read(ulaz);
   n := length(ulaz);

   for r:=1 to 5 do
      for s:=1 to 4*n+1 do
         izlaz[r] := izlaz[r] + '.';

   for i:=1 to n do
      crtaj( ulaz[i], 4*(i-1)+3, '#' );
   for i:=1 to n div 3 do
      crtaj( ulaz[3*i], 4*(3*i-1)+3, '*' );

   for r:=1 to 5 do
      writeln( izlaz[r] );
end.

File: Olympiad/COCI/official/2007/contest1_solutions/okviri.cpp

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 1 - Task OKVIRI
  Programming language C++
*/

#include <cstdio>
#include <cstring>
using namespace std;

char output[5][80];

int abs( int x ) { return x < 0 ? -x : x; }

void crtaj( char x, int c, char y ) {
   output[2][c] = x;
   for( int dr = -2; dr <= 2; ++dr )
      for( int dc = -2; dc <= 2; ++dc )
         if( abs(dr) + abs(dc) == 2 )
            output[2+dr][c+dc] = y;
}

int main( void ) {
   char s[16];
   scanf( "%s", s );

   int n = strlen( s );

   for( int r = 0; r < 5; ++r )
      for( int c = 0; c <= 4*n; ++c )
         output[r][c] = '.';

   for( int i = 0; i < n; ++i  ) crtaj( s[i], 4*i+2, '#' );
   for( int i = 2; i < n; i+=3 ) crtaj( s[i], 4*i+2, '*' );

   for( int r = 0; r < 5; ++r ) printf( "%s\n", output[r] );

   return 0;
}
